### ISSUE TITLE
Add optional 'render' stage

### ISSUE THREAD

**Issue by k** [GoogleContainerTools/skaffold#4356]
This is a feature request. I like the `skaffold render` command, but currently I am using it in a script to pipe helm output to a file and then apply a subset of the rendered manifests with kubectl based on the images that were built.

What would be better is if I could add a render step using helm to output the manifests natively in skaffold, then use the kubectl deployer with my custom flags that I use for various profiles to deploy. This should allow me to utilize more of the feature set of skaffold like testing and tailing logs.

Anyone else have this use case? For context I use ArgoCD for deployment which uses helm template under the hood. Unfortunately its not great for local development. This is where skaffold could fill that need but currently its hard to make use with helm without buying in to the whole helm deployment paradigm.

**Comment by tejal29** [comment]
@k i am not sure i understand what you mean by "render manifests natively in skaffold". 

`skaffold render` uses the `helm template` command to render manifests. 
https://github.com/GoogleContainerTools/skaffold/blob/fddc9544364b83766343219925bb9e819a043f79/pkg/skaffold/deploy/helm.go#L252
 

Do you mean, skaffold should first  only *render* using `helm template` and then deploy using `kubectl` deployer?

The team is looking into pre and post deploy hooks. This is something you could do as part of post render/deploy hook.

**Comment by k** [comment]
> Do you mean, skaffold should first only render using helm template and then deploy using kubectl deployer?

Yes thats exactly what I mean. We don't use helm to deploy so the helm deployer tries to do a helm install which fails because the manifests already exist in the cluster @tejal29

**Comment by k** [comment]
@tejal29 Will pre and post deploy hooks support helm/kustomize?

**Comment by tejal29** [comment]
@k, to understand you use case better, can you please share some part of your helm charts or skaffold logs `-v=debug`.

As far as we think,`helm template > manifests.yaml | helm install` should suffice. We would love to hear specific cases where it does not.

**Comment by GavinOsborn** [comment]
It sounds like @k and I have similar setups.
We use something that looks like `helm template | kubectl apply -f -` in our deployment pipelines to avoid installing `tiller` in our cluster. A bit of light Googling suggests this pattern isn't that uncommon either.

It would be nice if we could support this workflow with skaffold also for local development to catch edge cases that differ between the helm deployer and kubectl.

Last week we ran into an issue where our service deployed fine locally with skaffold but crashed and burned in our pipelines due to a difference between how the two systems handle casing (`loadBalancerIP` vs `loadBalancerIp`).

In truth it doesn't burn us that often but it would be cool with `Skaffold` supported this workflow out of the box.

(I'm aware that Helm 3 removes tiller, but #3798 prevents us from moving any time soon)

**Comment by briandealwis** [comment]
As you're aware, @GavinOsborn, #3798 has been fixed by #4513 and should be in the next release (RSN).

We'd definitely appreciate some more details about @k's situation though.

**Comment by tejal29** [comment]
duplicate of #5673
